clip_benchmark/cli.py

```diff
@@ -78,6 +89,37 @@ def run(args):
             device=args.device, 
             amp=args.amp
         )
+    elif args.task == "linear_probe":
+        # we also need the train split for linear probing.
+        train_dataset = build_dataset(
+            dataset_name=args.dataset, 
+            root=args.dataset_root, 
+            transform=transform, 
+            split='train', 
+            annotation_file=args.annotation_file,
+            download=True,
+        )
+        train_dataloader = torch.utils.data.DataLoader(
+            train_dataset, batch_size=args.batch_size, 
+            shuffle=False, num_workers=args.num_workers, 
+            collate_fn=collate_fn, pin_memory=True,
+        )
+        metrics = linear_probe.evaluate(
+            model,
+            train_dataloader, 
+            dataloader, 
+            args.fewshot_k,
+            args.batch_size,
+            args.num_workers,
+            args.fewshot_lr,
+            args.fewshot_epochs,
+            args.model + '-' + args.pretrained + '-' + args.dataset,
```

Comment:
Minor thing: `args.dataset` can have a `/` in it, when using `vtab/<name>` or `tfds/<name>` datasets. `os.mkdir(feature_dir)` in `evaluate` fails in that case, as it will try to create a subfolder. A small fix would be to just replace `/` with `_`, for `model_id`. Or replace os.mkdir by os.makedirs(feature_dir, exist_ok=True)`